---
title: "Computational approaches for circular RNA analysis"
date: 2019-01-01
publishDate: 2020-07-24T16:41:43.928003Z
authors: ["Tobias Jakobi", "Christoph Dieterich"]
publication_types: ["2"]
abstract: "Circular RNAs (circRNAs) are a recent addition to the expanding universe of RNA species and originate through back-splicing events from linear primary transcripts. CircRNAs show specific expression profiles with regards to cell type and developmental stage. Importantly, only few circRNAs have been functionally characterized to date. The detection of circRNAs from RNA sequencing data is a complex computational workflow that, depending on tissue and condition typically yields candidate sets of hundreds or thousands of circRNA candidates. Here, we provide an overview on different computational analysis tools and pipelines that became available throughout the last years. We outline technical and experimental requirements that are common to all approaches and point out potential pitfalls during the computational analysis. Although computational prediction of circRNAs has become quite mature in recent years, we provide a set of valuable validation strategies, in silico as well as in vitro-based approaches. In addition to circRNA detection via back-splicing junction, we present available analysis pipelines for delineating the primary sequence and for predicting possible functions of circRNAs. Finally, we outline the most important web resources for circRNA research. This article is categorized under: RNA Methods textgreater RNA Analyses in vitro and In Silico RNA Evolution and Genomics textgreater Computational Analyses of RNA. Â© 2019 Wiley Periodicals, Inc."
featured: false
publication: "*Wiley Interdisciplinary Reviews: RNA*"
tags: ["bioinformatics", "circular RNA", "RNA splicing", "software"]
doi: "10.1002/wrna.1528"
---

